CMakeLists.txt

```diff
@@ -1,30 +1,53 @@
 cmake_minimum_required(VERSION 3.16)
 set(OpenGL_GL_PREFERENCE GLVND)
 
+option(USE_VCPKG "Use vcpkg to handle some of the dependencies" OFF)
+if(DEFINED ENV{VCPKG_ROOT} AND NOT DEFINED CMAKE_TOOLCHAIN_FILE
+    OR USE_VCPKG)
+    set(OBE_USE_VCPKG ON)
+    set(CMAKE_TOOLCHAIN_FILE "$ENV{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake"
+        CACHE STRING "")
+endif()
+
+if(OBE_USE_VCPKG)
+    if(NOT DEFINED ENV{VCPKG_FEATURE_FLAGS})
+        set(ENV{VCPKG_FEATURE_FLAGS} "manifests") # manifests are required because of vcpkg.json
+    endif()
+        
+    if(NOT DEFINED ENV{VCPKG_DEFAULT_TRIPLET})
+        message(FATAL_ERROR
+            "Using vcpkg to build ObEngine without defining ${VCPKG_DEFAULT_TRIPLET}!! Aborting...")
+    endif()
+endif()
+
 project(ObEngine)
 
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_SOURCE_DIR}/cmake)
 include(engine_git)
 
 add_subdirectory(extlibs/bezier)
-add_subdirectory(extlibs/catch)
+
+if(NOT OBE_USE_VCPKG)
+    add_subdirectory(extlibs/catch)
+
+    # fmtlib is required by spdlog
+    add_subdirectory(extlibs/fmt)
+    add_subdirectory(extlibs/spdlog)
+
+    # lua is required by sol
+    add_subdirectory(extlibs/lua)
+    add_subdirectory(extlibs/sol3)
+endif()
+
 add_subdirectory(extlibs/dynamicLinker)
 add_subdirectory(extlibs/fswrapper)
 
-# fmtlib is required by spdlog
-add_subdirectory(extlibs/fmt)
-add_subdirectory(extlibs/spdlog)
-
 add_subdirectory(extlibs/vili)
 
-# lua is required by sol
-add_subdirectory(extlibs/lua)
-add_subdirectory(extlibs/sol3)
-
 # Audio library
 add_subdirectory(extlibs/soloud)
 
-if (APPLE OR USE_FILESYSTEM_FALLBACK)
+if (APPLE OR USE_FILESYSTEM_FALLBACK AND NOT OBE_USE_VCPKG)
     add_subdirectory(extlibs/tinydir)
     set(USE_FILESYSTEM_FALLBACK TRUE CACHE BOOL "Use filesystem fallback" FORCE)
```

Comment:
So if we use VCPKG the flag won't be set ?

Reply:
I added that here because Vcpkg has support for tinydir

Reply:
If `OBE_USE_VCPKG` is set to `TRUE` the line
```cmake
set(USE_FILESYSTEM_FALLBACK TRUE CACHE BOOL "Use filesystem fallback" FORCE)
```
will never be executed.
Then, in [src/Core/CMakeLists.txt#L38](https://github.com/Sygmei/ObEngine/blob/master/src/Core/CMakeLists.txt#L38) it will fail to do the appropriate actions.

Reply:
Fix coming right now